### ISSUE TITLE
selector-no-id reports interpolation inside Sass loops

### ISSUE THREAD

**Issue by kristerkari** [stylelint/stylelint#592]
stylelint is reporting interpolation inside of Sass loops as ID selectors.

This can be confirmed with unit tests:
https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-id/__tests__/index.js

```js
import scss from "postcss-scss"
const scssTestRule = ruleTester(rule, ruleName, {
  postcssOptions: { syntax: scss },
})

scssTestRule(undefined, tr => {
  tr.ok("@for $n from 1 through 10 { .page-#{$n} { color: blue; } }", "ignore sass interpolation inside @for")
  tr.ok("@each $n in $vals { .page-#{$n} { color: blue; } }", "ignore sass interpolation inside @each")
  tr.ok("@while $n < 10 { .page-#{$n} { color: blue; } }", "ignore sass interpolation inside @while")
})
```

```sh
# tests 8808
# pass  8805
# fail  3
```

**Comment by davidtheclark** [comment]
This is caused by postcss-selector-parser interpreting those `#{$n}` strings as ids. I'd bet that @ben-eb is not thrilled at the idea of making exceptions to the parser to deal with Sass-specific interpolation syntax. 

Is that right @ben-eb? Or do you see a way this can be handled?

So because of that, I'd say that if you want to use Sass interpolation in your selectors but also want to disallow id selectors, you might need to turn that rule off or include it in a post-Sass-compilation process.

**Comment by kristerkari** [comment]
> So because of that, I'd say that if you want to use Sass interpolation in your selectors but also want to disallow id selectors, you might need to turn that rule off or include it in a post-Sass-compilation process.

I was planning on using Atom linter and stylelint to do live linting for me (as I do with all other linting in my projects), so doing the checking for it in post-compilation would not work. 

I don't want to disable the rule because disallowing IDs in my `*.scss` files is the biggest reason to for me to use stylelint.

**Comment by ben-eb** [comment]
Yeah, you're right. :wink:

However you could make it part of your user code by testing for the curly braces either side of the ID node's value.

**Comment by kristerkari** [comment]
> However you could make it part of your user code by testing for the curly braces either side of the ID node's value.

That is also my solution https://github.com/stylelint/stylelint/pull/593

**Comment by davidtheclark** [comment]
Yeah, @kristerkari, I like your solution.

**Comment by davidtheclark** [comment]
Released in 3.2.0
